<?xml version="1.0"?>
<launch>

    
    <!-- <node pkg="rosbag" type="play" name="rosbag_play" args="clock $(arg bag_filename)"/> -->
    <!-- For Velodyne VLP-16 
    <param name="scan_line" value="16" />
    -->

    <!-- For Velodyne HDL-32  -->
    <param name="scan_line" value="32" />
   

    <!-- For Velodyne HDL-64 -->
    <!-- <param name="scan_line" value="64" /> -->


    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <param name="scan_period" value="0.1" /> 
    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="90.0" />
    <param name="min_dis" type="double" value="3.0" />
    <param name="topic" type="string" value="$(arg topic)" />
    <param name="k_new" type="string" value="$(arg k_new)"/>
    <param name="theta_p" type="string" value="$(arg theta_p)"/>
    <param name="theta_max" type="string" value="$(arg theta_max)"/>
    <param name="map_resolution" type="double" value="0.4"/>

    <!--- -->
    <node pkg="pfilter" type="PFilter_odom_estimation_node" name="PFilter_odom_estimation_node" output="screen"/>
    <node pkg="pfilter" type="PFilter_laser_processing_node" name="PFilter_laser_processing_node" output="screen"/>
    
    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find pfilter)/rviz/pfilter.rviz" />
    </group>

    
  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="velodyne" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
